staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -296,12 +296,23 @@ func canIdentifyUser(config restclient.Config) bool {
 
 // Namespace implements ClientConfig
 func (config *DirectClientConfig) Namespace() (string, bool, error) {
-	if config.overrides != nil && config.overrides.Context.Namespace != "" {
```

Comment:
I think this should go even deeper, down into the flag binding itself. We really only want this special handling to apply to the namespace flag itself (not necessarily the override struct), and there are other configs (like DeferredLoadingClientConfig that make use of overrides that would miss any special logic here).

Essentially, we want the data in the overrides struct to be good, not need fixing up. That could be done with a transforming string flag that lets us strip the namespace prefix prior to the value getting stored in the overrides struct - see https://github.com/liggitt/kubernetes/commit/e2478d3fb7c51d2615d10a1134a967a25eed7365 as an example